### PR TITLE
fix: Theming Selected new Font is reverted whenever Color is removed 

### LINKED ISSUE #15007: [Bug]: Theming :: Selected new Font is reverted whenever Color is removed
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

https://user-images.githubusercontent.com/91450662/177273879-44b3f0b5-94f1-4bb9-9d66-42594ff9fae1.mov

### Steps To Reproduce

1. Create a new App/Page - Select a theme
2. Observe the Font selected along with selected theme
3. **Now change the 'Font'**
4. **Now remove the value from color section to select/update a new value**
5. **Observe : Previously selected Font is also removed**
6. **Expected: Previous selection should not be removed - be it Font, Color, Shadow, Radius etc on change of any other property**

### Public Sample App

_No response_

### Version

Prod/Release v1.7.6

### PR DESCRIPTION
**Pull request by jsartisan**
Fixes #15007

/ok-to-test tags="@tag.Theme"

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/10733545727>
> Commit: dec21477152095d034f63a7e08d2719cb0687b12
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=10733545727&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.Theme`
> Spec:
> <hr>Fri, 06 Sep 2024 06:48:00 UTC

## Summary by CodeRabbit

- **New Features**
	- Enhanced theme validation for the multi-select widget, improving the robustness of theme-related tests.
	- Added new locators for featured themes to enhance testing capabilities.

- **Bug Fixes**
	- Reactivated a previously skipped test case to validate font consistency across the application.

- **Refactor**
	- Optimized color change handling in the ColorPicker component for better performance and reliability.
	- Refactored theme color control logic for improved maintainability and clarity.

- **Improvements**
	- Enhanced type safety in theme properties to prevent misconfigurations.
	- Improved testability of the ThemeCard and ThemeSelector components with new attributes for automated testing.

### CODE REVIEW

**Review by sagar-qa007 — APPROVED** [review]

File: app/client/cypress/e2e/Regression/ClientSide/ThemingTests/Theme_MultiSelectWidget_spec.js [hunk 1]
```diff
@@ -12,11 +12,22 @@ import {
...
   function () {
+    let themesSection = (sectionName, themeName) =>
+      "//*[text()='" +
+      sectionName +
+      "']/following-sibling::div//*[text()='" +
+      themeName +
+      "']";
+    let applyTheme = (sectionName, themeName) =>
+      themesSection(sectionName, themeName) +
+      "/parent::div/following-sibling::div[contains(@class, 't--theme-card')]//div[text()='Apply theme']";
```
[the remark this answers is not in the document]
Comment (sagar-qa007) [reply app/client/cypress/e2e/Regression/ClientSide/ThemingTests/Theme_MultiSelectWidget_spec.js:29]: @jsartisan can you check this suggestions as well please?

**Review by sagar-qa007 — CHANGES_REQUESTED** [review]

File: app/client/cypress/e2e/Regression/ClientSide/ThemingTests/Theme_MultiSelectWidget_spec.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (jsartisan) [reply app/client/cypress/e2e/Regression/ClientSide/ThemingTests/Theme_MultiSelectWidget_spec.js:29]: Ok let me update it.

**Comment by sagar-qa007** [comment]
@jsartisan  can we please start to use` /ci-test-limit-count ` now onwards? Benefit : No need to run command 2 times and saves a lot of time.

File: app/client/cypress/e2e/Regression/ClientSide/ThemingTests/Theme_MultiSelectWidget_spec.js [hunk 2]
```diff
@@ -107,33 +97,22 @@ describe(
...
-            .then((selectedBackgroudColor) => {
-              expect("rgba(0, 0, 0, 0)").to.equal(selectedBackgroudColor);
-              appSettings.ClosePane();
-            });
-        });
+      cy.get(
+        `${themelocator.featuredThemeSection} [data-testid='t--theme-card-Sunrise']`,
+      )
+        .click({ force: true })
+        .wait(1000);
```
Comment (sagar-qa007) [inline app/client/cypress/e2e/Regression/ClientSide/ThemingTests/Theme_MultiSelectWidget_spec.js:108]: Please remove wait.

**Review by sagar-qa007 — CHANGES_REQUESTED** [review]

**Review by znamenskii-ilia — DISMISSED** [review]

File: app/client/cypress/e2e/Regression/ClientSide/ThemingTests/Theme_MultiSelectWidget_spec.js [hunk 3]
```diff
@@ -107,33 +97,22 @@ describe(
...
 
-      //Publish the App and validate change of Theme across the app in publish mode
       deployMode.DeployApp();
-      cy.xpath("//div[@id='root']//section/parent::div").should(
-        "have.css",
-        "background-color",
-        "rgb(165, 42, 42)",
-      );
+
+      cy.get(".rc-select-selector").click({ force: true });
```
Comment (sagar-qa007) [inline app/client/cypress/e2e/Regression/ClientSide/ThemingTests/Theme_MultiSelectWidget_spec.js:112]: Please use `agHelper.GetNClick(locators._widgetInDeployed(draggableWidgets.VIDEO));`
Please .rc-select-selector add this in locator file.

**Review by sagar-qa007 — CHANGES_REQUESTED** [review]

**Review by sagar-qa007 — APPROVED** [review]

**Pull request merged**